src/visualization/types/Map/view.tsx

```diff
@@ -21,10 +32,23 @@ const GeoPlot: FC<Props> = ({result, properties}) => {
     detectCoordinateFields,
     mapStyle,
   } = properties
-  const tileServerConfiguration = {
-    tileServerUrl: '',
-    bingKey: '',
-  }
+
+  const [errorCode, setErrorCode] = useState<string>('')
+  const [mapToken, setMapToken] = useState<string>('')
+
+  useEffect(() => {
+    const getToken = async () => {
+      try {
+        const {token} = await getMapToken()
+        setMapToken(token)
+      } catch (err) {
+        setErrorCode(MapErrorStates.MapServiceUnavailable)
+      }
+    }
+    getToken()
+  }, [])
```

Comment:
I'm not a hooks expert, not sure if this empty array can be removed, or if we need to pass in `MapErrorStates`

Reply:
The empty array is not needed, I removed it .